safe-global/safe-react-apps
# feat(wallet-connect): switch to the safe-apps-provider under the hood

This came out as an attempt to fix #377

## What it solves
It switches the walletconnect Safe App to the `safe-apps-provider` package, so in case we change the handling of any of the RPC requests for the Safe Apps we can simply update it once in the SDK and bump the package version

## How to test it
Connect to your favourite app, transact with it and make sure everything is sound

## Why #377 wasn't solved
The problem is that in the case of walletconnect most of the wallet libraries are using the wallet provider only for WRITE requests, and READ requests are performed with a separate provider:
- Wagmi https://github.com/wagmi-dev/wagmi/blob/main/packages/core/src/connectors/walletConnect.ts#L133
- Onboard.js https://github.com/blocknative/web3-onboard/blob/v2-web3-onboard-develop/packages/walletconnect/src/index.ts#L244-L253
Therefore we cannot easily make it work with walletconnect. Further, I see three options:
1. Chase wallet libraries to adjust the behaviour and use the connector for read requests
2. Adjust .sendTransaction behaviour not to return safeTxHash right away and poll for the ethereum transaction hash
3. Do nothing and encourage apps to add Safe Apps support where this should work

I’m personally leaning towards 3) as the easiest option
